scripts/cluster_buildings.py

```diff
@@ -2,152 +2,144 @@
     central_points = []
     for i in range(kmeans.n_clusters):
-        cluster_points = points[kmeans.labels_ == i]
+        cluster_points = centroids_building[kmeans.labels_ == i]
         distances = np.linalg.norm(cluster_points - centroids[i], axis=1)
         central_point_idx = np.argmin(distances)
         central_points.append(cluster_points[central_point_idx])
 
-    # Create GeoJSON feature for each central point
     features = []
-    for i, central_point in enumerate(central_points):
-        feature = {
-            "type": "Feature",
-            "geometry": {"type": "Point", "coordinates": central_point.tolist()},
-            "properties": {"cluster": i},
-        }
-        features.append(feature)
-
-    # Create GeoJSON object with all central point features
-    geojson = {"type": "FeatureCollection", "features": features}
-
-    # Assign each building to its corresponding cluster based on k-means label
-    for i, label in enumerate(kmeans.labels_):
-        feature = data["features"][i]
-        if feature["geometry"]["type"] == "Point":
-            cluster_id = label
-            if "buildings" not in geojson["features"][cluster_id]["properties"]:
-                geojson["features"][cluster_id]["properties"]["buildings"] = []
-            geojson["features"][cluster_id]["properties"]["buildings"].append(
-                feature["properties"]["id"]
+    for i, row in enumerate(cleaned_buildings.itertuples()):
+        if row.geometry.type == "Polygon":
+            cluster_id = kmeans.labels_[i]
+            features.append(
+                {
+                    "properties": {
+                        "id": row.Index,
+                        "cluster_id": cluster_id,
+                        "tags_building": row.tags_building,
+                        "area_m2": row.area_m2,
+                    },
+                    "geometry": row.geometry,
+                }
             )
+    buildings_geodataframe = gpd.GeoDataFrame.from_features(features)
+    buildings_geodataframe.to_file(output_filepath)
+
 
-    # Write central point GeoJSON data with buildings to file
-    with open(output_filepath, "w") as f:
-        json.dump(geojson, f)
+def get_number_type_buildings(input_filepath, output_filepath):
+    buildings_geodataframe = gpd.read_file(input_filepath)
+
+    grouped_buildings = buildings_geodataframe.groupby("cluster_id")
+    clusters = np.sort(buildings_geodataframe["cluster_id"].unique())
+    counts = []
+    for cluster in clusters:
+        cluster_buildings = pd.DataFrame(grouped_buildings.get_group(cluster))
+        building_tag = cluster_buildings["tags_building"]
+        building_tag = pd.Series(building_tag)
+        count = building_tag.value_counts()
+        counts.append(count)
+    counts = pd.DataFrame(counts).fillna(0).astype(int)
+    counts["cluster"] = clusters
+    counts.set_index("cluster", inplace=True)
+    counts.to_csv(output_filepath)
```

Comment:
Sorry, I don't get exactly the goal of this.
What about
`buildings_geodataframe.cluster_id.value_counts()` ?
if you wish to also have the difference by tags_buildings, maybe:
`buildings_geodataframe.groupby("cluster_id").tags_building.value_counts()`

Reply:
Absolutely better, I had done it this way because I was not very familiar with values_count() and wanted to try to create a table to use in build_demand. 
I made a few changes to the build_demand rule to make it work with this proposal of yours, which is definitely better and lighter, thanks!

I would say I can compact everything in the function above , adding under the proposal you made above also the row:

building_class = buildings_geodataframe.groupby("cluster_id").tags_building.value_counts()

and also adding this last varible to the outputs